readers/api/transport.go

```diff
@@ -325,17 +323,14 @@ func authorize(ctx context.Context, token, key, chanID string) (err error) {
 	case token != "":
 		user, err := auth.Identify(ctx, &mainflux.Token{Value: token})
 		if err != nil {
-			e, ok := status.FromError(err)
-			if ok && e.Code() == codes.PermissionDenied {
-				return errors.Wrap(errUserAccess, err)
-			}
 			return err
 		}
+
+		if err := authorizeAdmin(ctx, user.Email); err == nil {
+			return nil
+		}
+
 		if _, err = things.IsChannelOwner(ctx, &mainflux.ChannelOwnerReq{Owner: user.Id, ChanID: chanID}); err != nil {
-			e, ok := status.FromError(err)
-			if ok && e.Code() == codes.PermissionDenied {
-				return errors.Wrap(errUserAccess, err)
```

Comment:
If `errUserAccess` is not used anymore remove the error. Same for `errThingAccess`. Also, check that both error can be distinguished and add the corresponding tests.